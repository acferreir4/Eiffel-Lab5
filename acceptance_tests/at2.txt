-- at2.txt: Simulate as many errors as possible
-- so that we can verify the correct error message
-- is shown to the user.
-- Error: names of players must be different
new_game("Simin", "Simin")
-- Error: name must start with A-Z or a-z
new_game("123", "456")
new_game("Andrew", "Dan")
-- Error: no such player
play("Eric", 1)
-- Error: not this player's turn
play("Dan", 2)
play("Andrew", 1)
-- Error: button already taken
play("Dan", 1)
play("Dan", 2)
play("Andrew", 4)
-- Error: finish this game first
play_again
play("Dan", 3)
play("Andrew", 7)
-- Error: game is finished
play("Dan", 9)
