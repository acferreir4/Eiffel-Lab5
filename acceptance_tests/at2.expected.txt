  ok:  => start new game
  ___
  ___
  ___
  0: score for "" (as X)
  0: score for "" (as O)
->new_game("Simin","Simin")
  names of players must be different:  => start new game
  ___
  ___
  ___
  0: score for "" (as X)
  0: score for "" (as O)
->new_game("123","456")
  name must start with A-Z or a-z:  => start new game
  ___
  ___
  ___
  0: score for "" (as X)
  0: score for "" (as O)
->new_game("Andrew","Dan")
  ok: => Andrew plays next
  ___
  ___
  ___
  0: score for "Andrew" (as X)
  0: score for "Dan" (as O)
->play("Eric",1)
  no such player: => Andrew plays next
  ___
  ___
  ___
  0: score for "Andrew" (as X)
  0: score for "Dan" (as O)
->play("Dan",2)
  not this player's turn: => Andrew plays next
  ___
  ___
  ___
  0: score for "Andrew" (as X)
  0: score for "Dan" (as O)
->play("Andrew",1)
  ok: => Dan plays next
  X__
  ___
  ___
  0: score for "Andrew" (as X)
  0: score for "Dan" (as O)
->play("Dan",1)
  button already taken: => Dan plays next
  X__
  ___
  ___
  0: score for "Andrew" (as X)
  0: score for "Dan" (as O)
->play("Dan",2)
  ok: => Andrew plays next
  XO_
  ___
  ___
  0: score for "Andrew" (as X)
  0: score for "Dan" (as O)
->play("Andrew",4)
  ok: => Dan plays next
  XO_
  X__
  ___
  0: score for "Andrew" (as X)
  0: score for "Dan" (as O)
->play_again
  finish this game first: => Dan plays next
  XO_
  X__
  ___
  0: score for "Andrew" (as X)
  0: score for "Dan" (as O)
->play("Dan",3)
  ok: => Andrew plays next
  XOO
  X__
  ___
  0: score for "Andrew" (as X)
  0: score for "Dan" (as O)
->play("Andrew",7)
  there is a winner: => play again or start new game
  XOO
  X__
  X__
  1: score for "Andrew" (as X)
  0: score for "Dan" (as O)
->play("Dan",9)
  game is finished: => play again or start new game
  XOO
  X__
  X__
  1: score for "Andrew" (as X)
  0: score for "Dan" (as O)
